webapp/src/main/resources/public/js/components/screenshots/ScreenshotViewerModal.js

```diff
@@ -96,7 +97,7 @@ class ScreenshotViewerModal extends React.Component {
                         <div className="branches-screenshotviewer-modal-delete-container">
                             {this.props.isDeleting ? <span className="glyphicon glyphicon-refresh spinning"/> :
                                 <OverlayTrigger trigger="click" placement="top" overlay={this.popover()}>
-                                    <Button bsStyle="danger" style={{fontSize: 11}}>
+                                    <Button bsStyle="danger" style={{fontSize: 11}} disabled={!AuthorityService.canEditScreenshots()}>
```

Comment:
for UI components that are stateless/exclusively use "props" I'd rather keep them like this instead of having a dependency on AuthorityService.

For others that state, I guess it is fine to have that AuthorityService depenedency

Reply:
OK, I have refactored it.